contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -393,23 +393,35 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
      */
     function sendSpecifiedReceipts(
         bytes32 originBlockchainID,
-        uint256[] calldata messageIDs,
+        bytes32[] calldata messageIDs,
         TeleporterFeeInfo calldata feeInfo,
         address[] calldata allowedRelayerAddresses
-    ) external senderNonReentrant returns (uint256) {
+    ) external senderNonReentrant returns (bytes32) {
+        bytes32 blockchainID_ = _initializeBlockchainID();
+
         // Iterate through the specified message IDs and create teleporter receipts to send back.
         TeleporterMessageReceipt[] memory receiptsToSend = new TeleporterMessageReceipt[](
                 messageIDs.length
             );
         for (uint256 i; i < messageIDs.length; ++i) {
-            uint256 receivedMessageID = messageIDs[i];
+            bytes32 receivedMessageID = messageIDs[i];
+
+            // Get the nonce for this message ID.
+            uint256 receivedMessageNonce = _receivedMessageNonces[receivedMessageID];
+            require(receivedMessageNonce != 0, "TeleporterMessenger: receipt not found");
```

Comment:
It would be helpful to emit the index of the missing receipt in the input array.

Reply:
I think this is actually more difficult that it's worth because it requires converting the `uint256` or `bytes32` to a string, and then doing string manipulation to format the error message.

Reply:
OTOH, requiring that all elements of the list are valid in order to send any receipts, especially when the list of pending receipts is always changing, is a less than ideal UI that may be difficult to integrate in an automated fashion. 

Could we instead silently drop invalid receipts, but still send the valid ones?

Reply:
(I recognize that this discussion isn't strictly in scope for this PR, so feel free to defer)

Reply:
I don't think this will be an issue either way really. 

The current state of the pending receipt queue doesn't affect `sendSpecifiedReceipts` here, because once a given message ID is received, its receipt will always be valid it to be sent via `sendSpecifiedReceipts`, whether or not the receipt has been sent in a message in the past. So automation shouldn't be too difficult. You can always check with `messageReceived` or `getRelayerRewardAddress` if a receipt is able to be sent for a message, and if it is, it will always be on a go forwards basis.

I think reverting in the event an invalid message ID is provided is more intuitive than silently dropping invalid ones, which would be likely to go unnoticed.